app/controllers/idv/document_capture_controller.rb

```diff
@@ -7,7 +7,7 @@ class DocumentCaptureController < ApplicationController
     include StepIndicatorConcern
 
     before_action :confirm_not_rate_limited, except: [:update]
-    before_action :confirm_step_allowed
+    before_action :confirm_step_allowed, unless: -> { allow_direct_ipp? }
```

Comment:
this does not  enforce that welcome and agreement pages were completed if IPP is enabled? 🤔  can we make this less lax?

Reply:
@amirbey, good suggestions, i was thinking add checking the referer, but this sounds better. Added check and tests.